src/components/multi_select_state.js

```diff
@@ -46,6 +49,16 @@ const withMultiSelectState = WrappedComponent =>
       if (this.props.searchValue !== nextProps.searchValue) {
         this.filterItems({ target: { value: nextProps.searchValue } });
       }
+
+      if (
+        this.props.searchSelectedItemsValue !==
+        nextProps.searchSelectedItemsValue
+      ) {
+        this.setState({ selectItemValue: nextProps.searchSelectedItemsValue });
+        this.filterSelectedItems({
+          target: { value: nextProps.searchSelectedItemsValue }
+        });
+      }
     }
 
     handleMultiSelection(index) {
```

Comment:
@MaksimDrobchak I think you missed this section. please check if it needs changes as well.